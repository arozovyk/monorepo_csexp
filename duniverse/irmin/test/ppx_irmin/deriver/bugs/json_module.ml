(* Ensure that the [Json] module in [Irmin.Type] doesn't shadow references to
   types contained in a different [Json] module.

   Disabled due to https://github.com/mirage/irmin/issues/923. *)

module Json = struct
  type t = string

  let t = Irmin.Type.string
end

type foo = { contents : Json.t } [@@deriving irmin]

let (_ : foo Irmin.Type.t) = foo_t
