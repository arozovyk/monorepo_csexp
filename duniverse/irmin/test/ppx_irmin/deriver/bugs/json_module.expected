module Json = struct type t = string
                     let t = Irmin.Type.string end
type foo = {
  contents: Json.t }[@@deriving irmin]
let foo_t =
  let open Irmin.Type in
    ((record "foo" (fun contents -> { contents })) |+
       (field "contents" Json.t (fun t -> t.contents)))
      |> sealr
let (_ : foo Irmin.Type.t) = foo_t
